### PR TITLE
Timeout fixes

### PR DESCRIPTION
**Pull request by burnsed**
1. Workaround docker daemon hangs
2. Fix bug where the script would wait again for a service that had already reached steady state.

### CODE REVIEW

File: src/rollover.py [hunk 1]
```diff
@@ -210,7 +210,8 @@ def map_instance_services(service_descriptions, task_descriptions):
             # only look at tasks with services (ignore instance startup tasks)
             ecs_id = utils.pull_instance_id(task['containerInstanceArn'])
             instance_services.setdefault(ecs_id, [])
-            instance_services[ecs_id].append(defs_to_services[def_arn])
+            if defs_to_services[def_arn] not in instance_services[ecs_id]:
+                instance_services[ecs_id].append(defs_to_services[def_arn])
```
Comment (nathanleiby) [inline src/rollover.py:214]: could use a `set`? (line 212)

this is totally fine though :)

**Comment by nathanleiby** [comment]
What action should the user take if docker daemon hangs?

**Comment by nathanleiby** [comment]
lgtm

**Comment by burnsed** [comment]
Updated to skip terminating machines with unresponsive docker daemons. Tested in rollout-test cluster.

File: src/rollover.py [hunk 2]
```diff
@@ -423,11 +447,21 @@ def main_rollover(args):
...
         print
 
-    if args.scale_down:
+    # Print the instances that need to be manually shutdown
+    if skipped_shutdown:
+        print "#"*80
+        print "The following instances could not be shutdown."
+        print "They likely still have tasks running on them:"
+        for instance in skipped_shutdown:
+            print instance
```
Comment (cleverworker) [inline src/rollover.py]: 👍

Comment (nathanleiby) [inline src/rollover.py:456]: 👍

**Comment by nathanleiby** [comment]
lgtm

**Pull request merged**